openpype/hosts/max/api/preview_animation.py

```diff
@@ -0,0 +1,305 @@
+        for key, value in nitrous_viewport_original.items():
+            setattr(viewport_setting, key, value)
+
+
+def _render_preview_animation_max_2024(
+        filepath, start, end, ext, viewport_options):
+    """Render viewport preview with MaxScript using `CreateAnimation`.
+    ****For 3dsMax 2024+
+    Args:
+        filepath (str): filepath for render output without frame number and
+            extension, for example: /path/to/file
+        start (int): startFrame
+        end (int): endFrame
+        viewport_options (dict): viewport setting options, e.g.
+            {"vpStyle": "defaultshading", "vpPreset": "highquality"}
+    Returns:
+        list: Created files
+    """
+    filepath = filepath.replace("\\", "/")
+    preview_output = f"{filepath}..{ext}"
+    frame_template = f"{filepath}.{{:04d}}.{ext}"
+    job_args = list()
+    default_option = f'CreatePreview filename:"{preview_output}"'
+    job_args.append(default_option)
+    frame_option = f"outputAVI:false start:{start} end:{end}"
+    job_args.append(frame_option)
+    for key, value in viewport_options.items():
+        if isinstance(value, bool):
+            if value:
+                job_args.append(f"{key}:{value}")
+        elif isinstance(value, str):
+            if key == "vpStyle":
+                if value == "Realistic":
+                    value = "defaultshading"
+                elif value == "Shaded":
+                    log.warning(
+                        "'Shaded' Mode not supported in "
+                        "preview animation in Max 2024.\n"
+                        "Using 'defaultshading' instead.")
+                    value = "defaultshading"
+                elif value == "ConsistentColors":
+                    value = "flatcolor"
+                else:
+                    value = value.lower()
+            elif key == "vpPreset":
+                if value == "Quality":
+                    value = "highquality"
+                elif value == "Customize":
+                    value = "userdefined"
+                else:
+                    value = value.lower()
+            job_args.append(f"{key}: #{value}")
+    auto_play_option = "autoPlay:false"
+    job_args.append(auto_play_option)
+    job_str = " ".join(job_args)
+    log.debug(job_str)
+    rt.completeRedraw()
+    rt.execute(job_str)
+    # Return the created files
+    return [frame_template.format(frame) for frame in range(start, end + 1)]
+
+
+def _render_preview_animation_max_pre_2024(
+        filepath, startFrame, endFrame, percentSize, ext):
+    """Render viewport animation by creating bitmaps
+    ***For 3dsMax Version <2024
+    Args:
+        filepath (str): filepath without frame numbers and extension
+        startFrame (int): start frame
+        endFrame (int): end frame
+        ext (str): image extension
+    Returns:
+        list: Created filepaths
+    """
+    # get the screenshot
+    percent = percentSize / 100.0
```

Comment:
Variables should not be camel case.
Also, size percentage was missing in docstring.

```suggestion
        filepath, startFrame, endFrame, percent_size, ext):
    """Render viewport animation by creating bitmaps
    ***For 3dsMax Version <2024
    Args:
        filepath (str): filepath without frame numbers and extension
        startFrame (int): start frame
        endFrame (int): end frame
        percentSize (float): render resolution multiplier by 100
            e.g. 100.0 is 1x, 50.0 is 0.5, 150.0 is 150%
        ext (str): image extension
    Returns:
        list: Created filepaths
    """
    # get the screenshot
    percent = percent_size / 100.0
```